docs_src/manuals/user_manual/interface.hrst

```diff
@@ -1,3 +1,5 @@
+TODO: Remove? Or are the instructions still valid for using cpp_bindgen separately?
+
```

Comment:
Whether or not this file should be removed is I think my main question. Some of it still applies since the text talks about fetch-contenting `cpp_bindgen`. However, the headers from gridtools that use `cpp_bindgen` are removed with this PR, so that part is definitely not going to work.

Do the nanobind headers completely replace all the cpp_bindgen functionality, or not fully?

Reply:
Sounds good to me. Would you keep the `fortran_array_view.hpp` and `fortran_array_adapter.hpp` headers where they are, but without any guarantees (they'll likely bitrot if all the tests that use them are removed; but it might not matter if gridtools isn't going to have many changes anymore)?

Reply:
Yes, not perfect. I would move the testing to cpp_bindgen (if there would be a use-case)...

Reply:
Restored the headers in https://github.com/GridTools/gridtools/pull/1821/commits/e387ed47c9e731841d17528123f52e983ce09ed0.